src/PrestaShopBundle/Command/AppendConfigurationFileHooksListCommand.php

```diff
@@ -0,0 +1,218 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Command;
+
+use DOMDocument;
+use Exception;
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\GridDefinitionFactoryInterface;
+use PrestaShop\PrestaShop\Core\Hook\Provider\HookByServiceIdsProviderInterface;
+use SimpleXMLElement;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerDebugCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * This command is used for appending the hook names in the configuration file.
+ */
+class AppendConfigurationFileHooksListCommand extends ContainerDebugCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('prestashop:update:configuration-file-hooks-listing')
+            ->setDescription('Appends configuration file hooks list')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        require $this->getContainer()->get('kernel')->getRootDir() . '/../config/config.inc.php';
+
+        //todo: check why not list hooks are not being listed.
+        $hookNames = $this->getHookNames();
+
+        $io = new SymfonyStyle($input, $output);
+
+        try {
+            $addedHooks = $this->appendHooksInConfigurationFile($hookNames);
+        } catch (Exception $e) {
+            $io->error($e->getMessage());
+        }
+
+        if (!empty($addedHooks)) {
+            $io->title('Hooks added to configuration file');
+            $io->note(sprintf('Total hooks added: %s', count($addedHooks)));
+            $io->listing($addedHooks);
+
+            return;
+        }
+
+        $io->note('No new hooks have been added to configuration file');
+    }
+
+    /**
+     * Gets all hooks names which need to be appended.
+     *
+     * @return string[]
+     */
+    private function getHookNames()
+    {
+        $container = $this->getContainer();
+        $containerBuilder = $this->getContainerBuilder();
```

Comment:
doing this defeats the purpose of the dependency injection, you may rely on a Compiler Pass instead.

Reply:
done :)